osbuild/objectstore.py

```diff
@@ -283,16 +277,14 @@ def __fspath__(self) -> os.PathLike:
 
 class ObjectStore(contextlib.AbstractContextManager):
     def __init__(self, store: PathLike):
-        self.store = store
-        self.objects = os.path.join(store, "objects")
-        self.refs = os.path.join(store, "refs")
+        self.cache = FsCache("osbuild", store)
         self.tmp = os.path.join(store, "tmp")
         os.makedirs(self.store, exist_ok=True)
         os.makedirs(self.objects, exist_ok=True)
-        os.makedirs(self.refs, exist_ok=True)
         os.makedirs(self.tmp, exist_ok=True)
         self._objs: Set[Object] = set()
         self._host_tree: Optional[HostTree] = None
+        self._stack: Optional[contextlib.ExitStack] = None
```

Comment:
You can avoid the `Optional` by creating the exit-stack here and using `_stack.close()` on `__exit__()`? `active` would then have to check for `self.cache._is_active()` or something else, though... Not sure. Just mentioning it here.

Reply:
I will do this now, maybe in a follow up we can make `cache.is_active()` or `cache.active` a public method / property.